hail/python/hail/matrixtable.py

```diff
@@ -3593,10 +3594,22 @@ def union_cols(self, other: 'MatrixTable') -> 'MatrixTable':
         The row fields in the resulting dataset are the row fields from the
         first dataset; the row schemas do not need to match.
 
-        This method performs an inner join on rows and concatenates entries
-        from the two datasets for each row. Only distinct keys from each
-        dataset are included (equivalent to calling :meth:`.distinct_by_row`
-        on each dataset first).
+        This method creates a :class:`.MatrixTable` which contains all columns
+        from both input datasets. The set of rows included in the result is
+        determined by the `row_join_type` parameter.
+
+        - With the default ``row_join_type=inner``, an inner join is performed
```

Comment:
We generally don't format the arg bit as code: should be something like:

    with the default value of ``'inner'``